Tests/OpenCombineTests/PublisherTests/FlatMapTests.swift

```diff
@@ -0,0 +1,539 @@
+        let upstreamSubscription = CustomSubscription()
+        let upstreamPublisher = CustomPublisherBase<Int, Never>(
+            subscription: upstreamSubscription)
+
+        let childSubscription = CustomSubscription()
+        let childPublisher = CustomPublisherBase<Int, Never>(
+            subscription: childSubscription)
+
+        let flatMap = upstreamPublisher.flatMap { _ in childPublisher }
+
+        var downstreamSubscription: Subscription?
+        let downstreamSubscriber = TrackingSubscriberBase<Int, Never>(receiveSubscription:
+        {
+            downstreamSubscription = $0
+            $0.request(.unlimited)
+        })
+
+        flatMap.subscribe(downstreamSubscriber)
+
+        XCTAssertEqual(upstreamPublisher.send(1), .none)
+
+        downstreamSubscription?.cancel()
+
+        XCTAssertEqual(upstreamSubscription.history.last!, .cancelled)
+        XCTAssertEqual(childSubscription.history.last!, .cancelled)
+    }
+
+    func testUpstreamDemandWithMaxPublishers() {
+        var upstreamDemand = Subscribers.Demand.unlimited
+        let upstreamSubscription = CustomSubscription(onRequest: { upstreamDemand = $0 })
+        let upstreamPublisher = CustomPublisherBase<Int, Never>(
+            subscription: upstreamSubscription)
+
+        let flatMap = upstreamPublisher.flatMap(maxPublishers: .max(2)) { Just($0) }
+
+        let downstreamSubscriber = TrackingSubscriberBase<Int, Never>(
+            receiveSubscription: { $0.request(.unlimited) })
+
+        flatMap.subscribe(downstreamSubscriber)
+
+        XCTAssertEqual(upstreamDemand, .max(2))
+    }
+
+    func testUpstreamDemandWithNoMaxPublishers() {
+        var upstreamDemand = Subscribers.Demand.max(666)
+        let upstreamSubscription = CustomSubscription(onRequest: { upstreamDemand = $0 })
+        let upstreamPublisher = CustomPublisherBase<Int, Never>(
+            subscription: upstreamSubscription)
+
+        let flatMap = upstreamPublisher.flatMap { Just($0) }
+
+        let downstreamSubscriber = TrackingSubscriberBase<Int, Never>(
+            receiveSubscription: { $0.request(.unlimited) })
+
+        flatMap.subscribe(downstreamSubscriber)
+
+        XCTAssertEqual(upstreamDemand, .unlimited)
+    }
+
+    func testChildDemand() {
+        let upstreamPublisher = PassthroughSubject<Void, Never>()
+
+        var childDemand: Subscribers.Demand?
+        let childSubscription = CustomSubscription(onRequest: { childDemand = $0 })
+        let childPublisher = CustomPublisherBase<Int, Never>(
+            subscription: childSubscription)
+
+        let flatMap = upstreamPublisher.flatMap { _ in childPublisher }
+
+        var downstreamSubscription: Subscription?
+        let downstreamSubscriber = TrackingSubscriberBase<Int, Never>(receiveSubscription:
+        {
+            downstreamSubscription = $0
+            $0.request(.max(2))
+        })
+
+        flatMap.subscribe(downstreamSubscriber)
+
+        upstreamPublisher.send(())
```

Comment:
Can you pleas use this overload here and below?
```suggestion
        upstreamPublisher.send()
```